python/pyspark/sql/connect/plan.py

```diff
@@ -2173,6 +2175,107 @@ def plan(self, session: "SparkConnectClient") -> proto.Relation:
         return plan
 
 
+class PythonUDTF:
+    """Represents a Python user-defined table function."""
+
+    def __init__(
+        self,
+        func: Type,
+        return_type: Union[DataType, str],
+        eval_type: int,
+        python_ver: str,
+    ) -> None:
+        self._func = func
+        self._name = func.__name__
+        self._return_type: DataType = (
+            UnparsedDataType(return_type) if isinstance(return_type, str) else return_type
+        )
+        self._eval_type = eval_type
+        self._python_ver = python_ver
+
+    def _parse_return_type(self, session: "SparkConnectClient") -> StructType:
+        if isinstance(self._return_type, UnparsedDataType):
+            parsed = session._analyze(
+                method="ddl_parse", ddl_string=self._return_type.data_type_string
+            ).parsed
+        else:
+            parsed = self._return_type
+
+        if not isinstance(parsed, StructType):
+            raise PySparkTypeError(
+                error_class="INVALID_UDTF_RETURN_TYPE",
+                message_parameters={"name": self._name, "return_type": f"{parsed}"},
+            )
```

Comment:
As this check is done in the server side, we don't need to check this in client side, then we can reduce one round-trip to parse the schema. `UnparsedDataType` will be handled in `SparkConnectPlanner.transformDataType`.